File: check_all_tables.py
import os
import psycopg2
from dotenv import load_dotenv

load_dotenv('backend/.env')

def check_counts():
    db_url = os.environ.get('DATABASE_URL')
    if not db_url:
        print("DATABASE_URL not found in .env")
        return

    try:
        conn = psycopg2.connect(db_url)
        cur = conn.cursor()

        tables = ['suppliers', 'accounting_vendors', 'store_suppliers', 'vendors']
        print("Checking record counts for vendor-related tables:")
        
        for table in tables:
            try:
                cur.execute(f"SELECT COUNT(*) FROM {table}")
                count = cur.fetchone()[0]
                print(f"{table}: {count} records")
            except Exception as e:
                print(f"{table}: Table not found or error: {e}")
                conn.rollback()

        print("\nChecking schema for suppliers table:")
        cur.execute("""
            SELECT column_name, data_type 
            FROM information_schema.columns 
            WHERE table_name = 'suppliers'
        """)
        columns = cur.fetchall()
        for col in columns:
            print(f"  {col[0]}: {col[1]}")

        print("\nChecking schema for accounting_ap_bills table:")
        cur.execute("""
            SELECT column_name, data_type 
            FROM information_schema.columns 
            WHERE table_name = 'accounting_ap_bills'
        """)
        columns = cur.fetchall()
        for col in columns:
            print(f"  {col[0]}: {col[1]}")

        cur.close()
        conn.close()

    except Exception as e:
        print(f"Connection error: {e}")

if __name__ == "__main__":
    check_counts()

File: check_all_tables.js
const { Pool } = require('pg');
require('dotenv').config({ path: 'backend/.env' });

const pool = new Pool({
    connectionString: process.env.DATABASE_URL,
    ssl: { rejectUnauthorized: false }
});

async function checkCounts() {
    try {
        const tables = ['suppliers', 'accounting_vendors', 'store_suppliers', 'vendors'];
        console.log("Checking record counts for vendor-related tables:");

        for (const table of tables) {
            try {
                const res = await pool.query(`SELECT COUNT(*) FROM ${table}`);
                console.log(`${table}: ${res.rows[0].count} records`);
            } catch (e) {
                console.log(`${table}: Table not found or error: ${e.message}`);
            }
        }

        console.log("\nChecking schema for suppliers table:");
        const schemaRes = await pool.query(`
      SELECT column_name, data_type 
      FROM information_schema.columns 
      WHERE table_name = 'suppliers'
    `);
        console.log(JSON.stringify(schemaRes.rows, null, 2));

        console.log("\nChecking schema for accounting_ap_bills table:");
        const billsRes = await pool.query(`
      SELECT column_name, data_type 
      FROM information_schema.columns 
      WHERE table_name = 'accounting_ap_bills'
    `);
        console.log(JSON.stringify(billsRes.rows, null, 2));

    } catch (err) {
        console.error(err);
    } finally {
        await pool.end();
    }
}

checkCounts();
